### DISCUSSION TITLE
Rename Identities?

### DISCUSSION THREAD

**Discussion by aleoyakas** [Flagsmith/flagsmith#2714] [Q&A] [answered]
We are currently mapping our user email addresses to our identities. We are doing this as we are wanting to easily distinguish users which users are which in the Flagsmith portal. On our platform, we currently allow users to change their email address, however, this means any traits the user had or any features explicitly set for them will be lost.

Is there a way to rename an Identity via the API to avoid this issue? 

If not, is there way to fetch a list traits belonging to the Identity via the API?

**Comment by dabeeeenster** [comment, marked as answer]
You can't rename identities currently. 

You can just grab the flags for an identity which will also return all their traits.

**Comment by aleoyakas** [comment]
When performing the following request:

- **Method:** Post
- **URL:** https://api.flagsmith.com/api/v1/identities/
- **Headers:** X-Environment-Key => Specified env key
- **Body:** 
```
{ "identifier":"bronze_user" }
```

I am receiving a the flags for the user, but the list of traits is empty, even though I have traits assigned to the user on the Flagsmith Portal.

If I update the body to include a new trait, e.g.
- **Body:**
```
{
  "identifier":"bronze_user",
  "traits": [
      {
            "trait_key": "my_trait_key",
            "trait_value": 123.5
        }
  ]
}
```

The new trait is present in the API response, however it is not visible in the Flagsmith Portal. When querying the API again, the new trait is present.

It seems that the Flagsmith Portal and the API have different data stores for a given environment - is that correct?

**Comment by dabeeeenster** [comment]
Is your project running against our Edge API? Can you amend the URL to https://edge.api.flagsmith.com/api/v1/identities/ ?

**Reply by aleoyakas** [reply]
I was just about to follow up with this - using the edge api endpoint works! Sorry for the confusion. Is there a reason we have 2 different endpoints? Or is it that we should only be using the edge endpoint if we're using the portal as well?

**Reply by dabeeeenster** [reply]
The Edge API is for SDKs to hit, the core API is for the dashboard and if you are driving the platform via CICD etc. So if youre doing stuff related to the SDKS (basically just getting flags, setting traits in identities etc) you should point to the edge api
